src/drt/src/dr/FlexDR_maze.cpp

```diff
@@ -3088,44 +3105,53 @@ void FlexDRWorker::routeNet_postAstarPatchMinAreaVio(
             net, bp, ep, gapArea, patchWidth, bpPatchStyle, epPatchStyle);
       }
       // init for next path
-      if (currIdx.z() < prevIdx.z()) {
-        currArea = getHalfViaEncArea(
-            prevIdx.z() - 1, true, net->getFrNet()->getNondefaultRule());
+      if (nextIdx.z() < currIdx.z()) {
+        // get the bottom layer box of the current via to initialize the area
+        // for the next shape
+        currArea
+            = getHalfViaEncArea(
+                  currIdx.z() - 1, true, net->getFrNet()->getNondefaultRule())
+              * 2;
```

Comment:
Is this correct if the next shape is a wire?

Reply:
I've tried to compensate for it in this line:
https://github.com/The-OpenROAD-Project/OpenROAD/pull/2581/files/b9d5dfea7a109560dcdc3f6625afec2c338280ee#diff-3b0ee5af511f4838323555d94ad93cc7da9df71a64bf049fa6b634da77978e64R3141

Reply:
What happens when there are multiple wire segments in series.  It seems "currArea /= 2;" would reduce the area of the previous segment not only the via.

Reply:
I've updated the code to perform `currArea /= 2;` only when the last object is not a wire. I believe this will prevent reducing the area of all the previous segments.

Reply:
My guess is this won't hurt anything.  Do you want to roll the dice or do another secure CI?

Reply:
We've waited a long time for this PR, and waiting a few more hours is fine. I'll start the secure-ci now.